airflow/models.py

```diff
@@ -1262,7 +1262,7 @@ def get_dagrun(self, session):
         return dr
 
     @provide_session
-    def run(
+    def set_running_state_before_execution(
```

Comment:
It is coupled in run, but in different contexts you may not want to execute them together.